File: backend/src/main/kotlin/org/globe42/dao/ChargeCategoryDao.java
package org.globe42.dao;

import java.util.Optional;

import org.globe42.domain.ChargeCategory;
import org.springframework.data.jpa.repository.JpaRepository;

/**
 * DAO for the {@link ChargeCategory} entity
 * @author JB Nizet
 */
public interface ChargeCategoryDao extends JpaRepository<ChargeCategory, Long> {
    boolean existsByName(String name);
    Optional<ChargeCategory> findByName(String name);
}

File: backend/src/main/kotlin/org/globe42/dao/ChargeCategoryDao.kt
package org.globe42.dao

import org.globe42.domain.ChargeCategory
import org.springframework.data.jpa.repository.JpaRepository

/**
 * DAO for the [ChargeCategory] entity
 * @author JB Nizet
 */
interface ChargeCategoryDao : JpaRepository<ChargeCategory, Long> {
    fun existsByName(name: String): Boolean
    fun findByName(name: String): ChargeCategory?
}
